Gcpe.Hub.API.Tests/ControllerTests/SocialMediaControllerTests.cs

```diff
@@ -0,0 +1,292 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoMapper;
+using FluentAssertions;
+using Gcpe.Hub.API.Controllers;
+using Gcpe.Hub.API.Data;
+using Gcpe.Hub.API.ViewModels;
+using Gcpe.Hub.Data.Entity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit;
+
+namespace Gcpe.Hub.API.Tests.ControllerTests
+{
+    public class SocialMediaControllerTests
+    {
+        private Mock<ILogger<SocialMediaController>> logger;
+        private HubDbContext context;
+        private IMapper mapper;
+        private SocialMediaController controller;
+
+        public SocialMediaControllerTests()
+        {
+            var options = new DbContextOptionsBuilder<HubDbContext>()
+                      .UseInMemoryDatabase(Guid.NewGuid().ToString())
+                      .Options;
+            this.context = new HubDbContext(options);
+            this.logger = new Mock<ILogger<SocialMediaController>>();
+            var mockMapper = new MapperConfiguration(cfg =>
+            {
+                cfg.AddProfile(new MappingProfile());
+            });
+            this.mapper = mockMapper.CreateMapper();
+            this.controller = new SocialMediaController(context, logger.Object, mapper);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(3)]
+        [InlineData(10)]
+        public void GetAll_ShouldReturnSuccessAndSorted(int postCount)
+        {
+            for (var i = 0; i < postCount; i++)
+            {
+                var post = TestData.CreateSocialMediaPost("http://facebook.com/post/123");
+                post.Id = Guid.NewGuid();
+                post.SortOrder = postCount - i;
+                context.SocialMediaPost.Add(post);
+            }
+            context.SaveChanges();
+
+            var result = controller.GetAll();
+            var okResult = result as ObjectResult;
+
+            okResult.Should().BeOfType<OkObjectResult>();
+            okResult.Should().NotBeNull();
+
+            var models = okResult.Value as IList<SocialMediaPostViewModel>;
+            models.Should().NotBeNull();
+            models.Count().Should().Be(postCount);

```

Comment:
How does this pass?

Reply:
for 0: models = [] (count = 0 = postCount)
for 1: models = [post1] (count = 1 = postCount)
for 3: models = [post1, post2, post3] (count = 3 = postCount)
etc